client/components/StarRating.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { FaStar } from 'react-icons/fa';
+
+import '../css/StarRating.scss';
+
+const Span = styled.span`
+  color: ${({ backgroundColor }) => backgroundColor};
+  display: inline-block;
+  font-size: ${({ size }) => size}px;
+  margin: 0 5px;
+  position: relative;
+
+  &:after {
+    font-family: "Font Awesome 5 Free";
+    font-weight: 900;
+    content: "\f005";
+    position: absolute;
+    left: 0;
+    top: 0;
+    width: ${({ percentage }) => percentage}%;
```

Comment:
So the width property determines the amount of the star filled?

Reply:
Correct. There's a small example here https://jsfiddle.net/vco9r2rt/3/